src/node_process_methods.cc

```diff
@@ -423,124 +416,119 @@ static void ReallyExit(const FunctionCallbackInfo<Value>& args) {
-  // so this function instead fills in an Uint32Array with 3 entries,
-  // to avoid any integer overflow possibility.
-  // The first two entries contain the second part of the value
-  // broken into the upper/lower 32 bits to be converted back in JS,
-  // because there is no Uint64Array in JS.
-  // The third entry contains the remaining nanosecond part of the value.
-  static void NumberImpl(FastHrtime* receiver) {
-    uint64_t t = uv_hrtime();
-    uint32_t* fields = static_cast<uint32_t*>(receiver->backing_store_->Data());
-    fields[0] = (t / NANOS_PER_SEC) >> 32;
-    fields[1] = (t / NANOS_PER_SEC) & 0xffffffff;
-    fields[2] = t % NANOS_PER_SEC;
-  }
+void BindingData::AddMethods() {
+  env()->SetFastMethod(object(), "hrtime", SlowNumber, &fast_number_);
+  env()->SetFastMethod(object(), "hrtimeBigInt", SlowBigInt, &fast_bigint_);
+}
 
-  static void FastNumber(Local<Value> receiver) {
-    NumberImpl(FromV8Value(receiver));
-  }
+void BindingData::RegisterExternalReferences(
+    ExternalReferenceRegistry* registry) {
+  registry->Register(SlowNumber);
+  registry->Register(SlowBigInt);
+  registry->Register(FastNumber);
+  registry->Register(FastBigInt);
+  registry->Register(fast_number_.GetTypeInfo());
+  registry->Register(fast_bigint_.GetTypeInfo());
+}
 
-  static void SlowNumber(const FunctionCallbackInfo<Value>& args) {
-    NumberImpl(FromJSObject<FastHrtime>(args.Holder()));
-  }
+BindingData* BindingData::FromV8Value(Local<Value> value) {
+  Local<Object> v8_object = value.As<Object>();
+  return static_cast<BindingData*>(
+      v8_object->GetAlignedPointerFromInternalField(BaseObject::kSlot));
+}
 
-  static void BigIntImpl(FastHrtime* receiver) {
-    uint64_t t = uv_hrtime();
-    uint64_t* fields = static_cast<uint64_t*>(receiver->backing_store_->Data());
-    fields[0] = t;
-  }
+void BindingData::MemoryInfo(MemoryTracker* tracker) const {
+  tracker->TrackField("array_buffer", array_buffer_);
+}
 
-  static void FastBigInt(Local<Value> receiver) {
-    BigIntImpl(FromV8Value(receiver));
-  }
+// This is the legacy version of hrtime before BigInt was introduced in
+// JavaScript.
+// The value returned by uv_hrtime() is a 64-bit int representing nanoseconds,
+// so this function instead fills in an Uint32Array with 3 entries,
+// to avoid any integer overflow possibility.
+// The first two entries contain the second part of the value
+// broken into the upper/lower 32 bits to be converted back in JS,
+// because there is no Uint64Array in JS.
+// The third entry contains the remaining nanosecond part of the value.
+void BindingData::NumberImpl(BindingData* receiver) {
+  // Make sure we don't accidentally access buffers wiped for snapshot.
+  CHECK(!receiver->array_buffer_.IsEmpty());
+  uint64_t t = uv_hrtime();
+  uint32_t* fields = static_cast<uint32_t*>(receiver->backing_store_->Data());
+  fields[0] = (t / NANOS_PER_SEC) >> 32;
+  fields[1] = (t / NANOS_PER_SEC) & 0xffffffff;
```

Comment:
```suggestion
  fields[1] = (t / NANOS_PER_SEC) & 0xffff'ffff;
```
Shouldn't we use a separator for clarity?

Reply:
That might be nicer, but this is just moving existing code around and I think it might be better to just leave any unrelated changes to a different PR.